e2e/playground.py

```diff
@@ -0,0 +1,2135 @@
+    loc_choice_labels: Locator
+
+    def expect_choice_labels(
+        self,
+        labels: typing.Union[PatternOrStr, typing.List[PatternOrStr]],
+        *,
+        timeout: Timeout = None,
+    ) -> None:
+        playwright_expect(self.loc_choice_labels).to_have_text(
+            labels,
+            timeout=timeout,
+        )
+
+    def expect_inline(self, inline: bool, *, timeout: Timeout = None) -> None:
+        if inline:
+            playwright_expect(self.loc_container).to_have_class(
+                "shiny-input-container-inline", timeout=timeout
+            )
+        else:
+            playwright_expect(self.loc_container).not_to_have_class(
+                "shiny-input-container-inline", timeout=timeout
+            )
+
+    def _expect_locator_values_in_list(
+        self: _InputWithContainerP,
+        el_type: str,
+        arr_name: str,
+        # TODO; support patterns?
+        arr: typing.List[str],
+        *,
+        is_checked: typing.Optional[bool] = None,
+        timeout: Timeout = None,
+        key: str = "value",
+    ) -> None:
+        # Make sure the locator has all of arr
+
+        # Make sure the locator has len(uniq_arr) input elements
+        assert len(arr) == len(list(dict.fromkeys(arr))), f"`{arr_name}` must be unique"
+
+        if is_checked is None:
+            is_checked_str = ""
+        elif is_checked:
+            is_checked_str = ":checked"
+        else:
+            NotImplementedError("`is_checked = FALSE` is not verified yet")
+            is_checked_str = ":not(:checked)"
+
+        # Find all items in set
+        loc_container = self.loc_container
+        for (item, i) in zip(arr, range(len(arr))):
+            # # Simple approach if position wasn't needed
+            # has_locator = self.page.locator(
+            #     f"{el_type}[{key}='{item}']{is_checked_str}"
+            # )
+
+            has_locator = self.page.locator(f"{el_type}{is_checked_str}")
+            # Go up one element as CSS does not have "self" selector
+            # We can leverage this as the HTML structure is very nested.
+            # This approach will break if there are matching siblings
+            has_locator = has_locator.nth(i).locator("..")
+            # Find direct child that matches selector
+            has_locator = has_locator.locator(
+                f"> {el_type}[{key}='{item}']{is_checked_str}"
+            )
+
+            loc_container = loc_container.locator(
+                # Return self
+                "xpath=.",
+                has=has_locator,
+            )
+        # Make sure other items are not in set
+        loc_inputs = loc_container.locator(f"{el_type}{is_checked_str}")
+        # TODO-barret; Look into adding a try-catch around this and then performing the locator check using multiple expectations to get better error messages
+        playwright_expect(loc_inputs).to_have_count(len(arr), timeout=timeout)
```

Comment:
This will fail if `arr` is empty. Need some earlier logic to prevent us from getting here.